src/kv/encryptor.h

```diff
@@ -91,6 +91,8 @@ namespace kv
      * @param[out]  plain             Decrypted plaintext
      * @param[in]   version           Version used to retrieve the corresponding
      * encryption key
+     * @param[in]   is_historical     If set, only consider encryption keys
+     * that have not been compacted away (used to decrypt historical queries)
```

Comment:
I don't understand why we need this flag. Shouldn't we just treat the secrets as a cache? Why do we need to communicate this to decrypt?